test/unit/notify/send-email.test.js

```diff
@@ -102,11 +102,11 @@ describe('send email', () => {
     expect(notifyClientMockInstance.sendEmail).toHaveBeenCalledTimes(1)
   })
 
-  test('should call notifyClient.sendEmail with correct parameters when a valid message is received', async () => {
```

Comment:
The original test case needs to remain, however you will need to change the `defaultReference` from an empty string to a mockImplementation result of `uuidv4()`

You can then add back in the test ensuring that  when `sendEmail` is called w/out a `reference` param it is not passed to `notifyClientMockInstance.sendEmail` as an empty string

Reply:
The next test (line 115 to 123) has implemented your suggestion

Reply:
The next test covers the scenario when you supply a `reference` parameter.

So what we need to test are 2 scenarios:

1. test the scenarios when only the required params `context` and `message` are given
    **note:** this will mean that `reference` will default
2. test the scenarios when the optional params `emailAddresses` and `reference` are also given


The previous test covered scenario 1, stating that the expected `reference` when none was supplied was `defaultReference` which was `''`.

We've changed it that now that default `reference` is a `uuid()`. Therefore we still need to keep this test but now must modify what `defaultReference` is (previously it was `''`, but now it will be `mockReference`) and we need to 
mock what `uuid` will return when it is called, something like this `uuid.mockImplementation(() => mockReference)` -- this will ensure that our mocked `references` match for testing purposes.


Happy to havva 5 min call to better explain if needs be